components/src/core/drawer/drawer.component.ts

```diff
@@ -0,0 +1,75 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    ViewEncapsulation,
+    Input,
+    Output,
+    EventEmitter,
+    OnInit,
+    ChangeDetectorRef,
+} from '@angular/core';
+import { DrawerService } from './drawer.service';
+
+export type VariantType = 'permanent' | 'persistent' | 'temporary';
+
+@Component({
+    selector: 'pxb-drawer',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    template: `
+        <div class="pxb-drawer">
+            <!-- Drawer is responsible for managing the styles between the 4 subsections -->
+            <ng-content select="pxb-drawer-header"></ng-content>
+            <div (mouseenter)="openDrawer()" (mouseleave)="closeDrawer()">
+                <ng-content select="pxb-drawer-subheader"></ng-content>
+                <ng-content select="pxb-drawer-body"></ng-content>
+                <ng-content select="pxb-drawer-footer"></ng-content>
+            </div>
+        </div>
+    `,
+    styleUrls: ['./drawer.component.scss'],
+})
+export class DrawerComponent implements OnInit {
+    @Input() variant: VariantType;
+    @Input() variantDrawerHandler: boolean;
+    @Output() drawerOpenChange: EventEmitter<boolean> = new EventEmitter();
+    drawerOpen: boolean;
```

Comment:
Where does drawerOpen actually get used to change the appearance of the drawer from navigation rail to fully opened (for persistent)?